src/debug.rs

```diff
@@ -0,0 +1,139 @@
+//! Functions for interacting with the postponed jobs API.
+
+use std::ffi::c_void;
+
+use rb_sys::rb_postponed_job_register;
+
+/// Registers a function to be executed when the Ruby VM is safe to access
+/// (i.e. inside a signal handler or during a GC run).
+///
+/// ```rust
+/// # let _cleanup = unsafe { magnus::embed::init() };
+///
+/// use magnus::{debug::postponed_job_register, Value};
+/// use std::sync::atomic::{AtomicUsize, Ordering};
+///
+/// static COUNTER: AtomicUsize = AtomicUsize::new(0);
+///
+/// let increment_counter = || {
+///   COUNTER.fetch_add(1, Ordering::SeqCst);
+/// };
+///
+/// match postponed_job_register(increment_counter) {
+///     Ok(()) => println!("Postponed job registered!"),
+///     Err(e) => println!("Failed to register postponed job: {}", e),
+/// };
+///
+/// assert_eq!(0, COUNTER.load(Ordering::SeqCst));
+/// let _: Value = magnus::eval!("Thread.pass").unwrap();
+/// assert_eq!(1, COUNTER.load(Ordering::SeqCst));
+/// ```
+pub fn postponed_job_register<F: FnOnce()>(func: F) -> Result<(), PostponedJobError> {
+    unsafe extern "C" fn trampoline<F: FnOnce()>(f: *mut c_void) {
+        let f: Box<F> = Box::from_raw(f as *mut F);
```

Comment:
I think `F` needs `Send + 'static` bounds so that any variables it captures are guaranteed to last until it's called.

But maybe it could just be a `fn()` function pointer? That way we wouldn't need to box it for the trampoline.